src/main/java/com/redhat/cloud/policies/app/auth/RbacFilter.java

```diff
@@ -99,8 +99,8 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
     private RbacRaw getRbacResult() {
         RbacRaw result;
         long t1 = System.currentTimeMillis();
-        Span span = tracer.buildSpan("getRBac").start();
-        try (Scope ignored = tracer.scopeManager().activate(span)) {
+        Span span = tracer.spanBuilder("getRBac").startSpan();
+        try (Scope ignored = span.makeCurrent()) {
             result = rbacClient.getRbacInfo(user.getRawRhIdHeader());
         } catch (Throwable e) {
             Log.warn("RBAC call failed", e);
```

Comment:
you should record the exception in the span:
````java
span.recordException(e);
````

Reply:
Added